### DISCUSSION TITLE
Should we release 0.71.12?

### DISCUSSION THREAD

**Discussion by kelset** [reactwg/react-native-releases#73] [Patches]
## Released, please refer to #78 

----

Current Release: [0.71.11](https://github.com/facebook/react-native/releases/tag/v0.71.11)

Conversations on this thread are limited:

- [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking).
- [qualified cherry-pick requests](https://reactnative.dev/contributing/release-faq#what-is-a-qualified-pick-request) of commits on main that [did not make the previous patch version](https://reactnative.dev/contributing/release-faq#how-do-i-know-if-my-fixfeature-is-in-a-certain-release).

Please include a link to the specific commit on main to be cherry-picked, for example: [facebook/react-native@bd2b7d6](https://github.com/facebook/react-native/commit/20b0eba581a00e5e7e300f6377379b836617c147)

In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.
If the issue is a [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking), please reference the issue here.

---

#### List of qualified picks

1. https://github.com/facebook/react-native/commit/cd56347dca4e948f5038643bcd804c41f037727a
2. https://github.com/facebook/react-native/pull/37876
3. https://github.com/facebook/react-native/pull/37993

#### Local commits to backport to main

1.

**Comment by kelset** [comment]
from @AlexanderEggers and @NickGerleman:

> It would be great to backport https://github.com/facebook/react-native/commit/3681df287835f5467a2ad5afe950eae16b95fd8b to fix https://github.com/facebook/react-native/issues/37797.

PR for 0.71 branch -> https://github.com/facebook/react-native/pull/37876

**Comment by Saadnajmi** [comment]
Could we pick https://github.com/facebook/react-native/commit/8b1bf058c4bcbf4e5ca45b0056217266a1ed870c ?

**Reply by Saadnajmi** [reply]
Cc @huntie

**Reply by kelset** [reply]
based on what @tido64 is saying, it looks like this would force folks to update to Xcode 14.3 so technically it'd be a breaking change... we need to be careful.

**Reply by huntie** [reply]
I think we should look to concretely define React Native's compatibility model relating to Xcode versions. 14.3 is the current (non beta) release of Xcode and IMO makes this not a breaking framework change (not that that was said) — but a compatibility note we should consistently highlight in release notes and our website. cc @cipolleschi @cortinico  @lunaleaps @rickhanlonii

**Reply by kelset** [reply]
github actions' macos-latest image is still on xcode 14.2 -> https://github.com/actions/runner-images/blob/main/images/macos/macos-12-Readme.md#tools

so we'd literally be breaking any CI that relies on it.

**Reply by kelset** [reply]
btw I totally agree that we should define a better "compatibility model" between tooling and releases, but even that in case we'd have to associate minors with toolings versions/minimum requirements. It was discussed already a very long time ago ([look at this comment](https://github.com/react-native-community/discussions-and-proposals/issues/17#issuecomment-446261269), I still think about it from time to time 😅) but in this specific case, for 72 and 71, adding this commit would by all means be a breaking change because it would force a change in users requirements  (and again, would literally make the new releases incompatible with at least GitHub actions --> big no no).

**Reply by tido64** [reply]
For reference, this is what we had to do in RNTA: https://github.com/microsoft/react-native-test-app/pull/1459/files

- Use `macos-13` agents
- Manually run `xcode-select` to set the correct Xcode version

Update: Just discovered that some `macos-13` agents are having issues executing XCTests. It's been fixed, but the new image may take days to propagate to all agents.

**Reply by kelset** [reply]
I've been talking on the side with @Saadnajmi about this: maybe what we can do is that we keep this cherry-pick request "frozen" across patches ([similar to what I do](https://github.com/reactwg/react-native-releases/discussions/73#discussioncomment-6176798) when there's some work we couldn't handle in a specific patch) until both at least CircleCI and GitHub Actions default their macos images (the `-latest` ones) to versions that have xcode >=14.3, and at that point we put it in an actual patch release. WDYT?

**Reply by cortinico** [reply]
> but a compatibility note we should consistently highlight in release notes and our website. cc @cipolleschi @cortinico @lunaleaps @rickhanlonii

For context: on Android, the AGP version creates the lowerbound of the Android Studio version that is supported (as described here https://developer.android.com/studio/releases#api-level-support).
So for example 0.71 shipped with AGP 7.3.1 which means Android Studio Dolphin or later releases. 0.72 ships with AGP 7.4.x == Android Studio Electric Eel or later. Etc.

We can totally add compatibility notes somewhere

**Reply by Saadnajmi** [reply]
With Apple in my experience, it's generally harder to enforce compatibility beyond "Use the latest Xcode version". We're kind of forced to use the latest version of Xcode as soon as it comes out and it's generally a flurry of "update CI everywhere and go make fixes". With other projects I've been involved in, I don't think we've ever made any sort of "We support this many versions of Xcode" kind of agreement, only "We fixed the bugs we found with the latest Xcode, and here are the versions of iOS or macOS we support". That's also why I wrapped the call in OS check, that's fairly standard practice.

In this case, I understand that we can't really force all projects to update their CI to not-yet-the-standard-image in a patch release. I'm happy to revisit this change in the future, hopefully too many users aren't impacted.

**Reply by liamjones** [reply]
Possibly relevant to the discussion; Apple currently mandate only Xcode 14.1+ for uploads to the store (as of April this year): https://developer.apple.com/news/upcoming-requirements/?id=04252023a

**Reply by Saadnajmi** [reply]
I'll see if I can make a version that uses compile time checks to be backwards compatible, but I think mostly likely this will not get backported.

**Reply by Saadnajmi** [reply]
Found in `Availability.h`. Might give us a compile time check we can use. 
```
    It is also possible to use the *_VERSION_MIN_REQUIRED in source code to make one
    source base that can be compiled to target a range of OS versions.  It is best
    to not use the _MAC_* and __IPHONE_* macros for comparisons, but rather their values.
    That is because you might get compiled on an old OS that does not define a later
    OS version macro, and in the C preprocessor undefined values evaluate to zero
    in expresssions, which could cause the #if expression to evaluate in an unexpected
    way.
    
        #ifdef __MAC_OS_X_VERSION_MIN_REQUIRED
            // code only compiled when targeting Mac OS X and not iPhone
            // note use of 1050 instead of __MAC_10_5
            #if __MAC_OS_X_VERSION_MIN_REQUIRED < 1050
                // code in here might run on pre-Leopard OS
            #else
                // code here can assume Leopard or later
            #endif
        #endif
```

**Comment by kelset** [comment]
https://github.com/facebook/react-native/pull/37993

**Comment by hannojg** [comment]
This is a workaround for android API 33+ that prevents ANRs from happening: https://github.com/facebook/react-native/commit/90186cd9b71bc6ffb593448c9bb5a3df66b3a0c6

 Would appreciate to see it in a patch version 🙌

**Reply by liamjones** [reply]
This was reverted: https://github.com/facebook/react-native/pull/37913#event-9604118424

**Reply by kelset** [reply]
good catch: there are now two PRs:
* https://github.com/facebook/react-native/pull/38071
* https://github.com/facebook/react-native/pull/38073

**Comment by evelant** [comment]
Can https://github.com/facebook/react-native/pull/38005/commits/e3abbc26c051bbd241effd037c7b0afb6902e98e get into the next patch release? It's a serious DX drag to have the dev app freeze up all the time due to logbox crashing.

**Reply by joe-sam** [reply]
Actual commit is probably merged @ Prevent LogBox from crashing on long messages (https://github.com/facebook/react-native/pull/38005) 
LogBox related fixes should after some nightly testing be immediately backported to all supported latest RN releases as these are definitely show-stopping bug fixes and can save loads of time for developers.
